modules/apps/portal-workflow/portal-workflow-kaleo-service/src/main/java/com/liferay/portal/workflow/kaleo/internal/upgrade/v3_1_1/KaleoNotificationUpgradeProcess.java

```diff
@@ -48,16 +48,19 @@ protected void doUpgrade() throws Exception {
 			ResultSet resultSet = preparedStatement1.executeQuery();
 
 			while (resultSet.next()) {
-				String[] notificationTypes = Stream.of(
-					StringUtil.split(resultSet.getString("notificationTypes"))
-				).filter(
-					notificationType -> !Objects.equals(notificationType, "im")
-				).filter(
-					notificationType -> !Objects.equals(
-						notificationType, "private-message")
-				).toArray(
-					String[]::new
-				);
+				String[] notificationTypes = TransformUtil.transform(
+					StringUtil.split(resultSet.getString("notificationTypes")),
+					notificationType -> {
+						if (!Objects.equals(notificationType, "im") &&
+							!Objects.equals(
+								notificationType, "private-message")) {
+
+							return notificationType;
+						}
+
+						return null;
```

Comment:
@JanisZhang Please return null in shortcut. By the way, please review your transformer usages, and put the lambda variable second in Objects.equals.